bazel/scripts/setup_bazel_bash_completion.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+################################################################################
+# Copyright 2022 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+########################################################
+# Configures Bazel Bash-completion for the invoking user
+########################################################
+
+set -euo pipefail
+
+function log() {
+  echo "$@" >&2
+}
+
+function generate_completion() {
+  # outputs the completion script to stdout
+  #
+  # inspired by https://github.com/bazelbuild/bazel/blob/5f3f59ba367158b6c2e811f68cc946537b9b74d6/scripts/BUILD#L6-L26
+  # which is referenced in https://bazel.build/install/completion
+  curl https://raw.githubusercontent.com/bazelbuild/bazel/release-5.3.0/scripts/bazel-complete-header.bash
```

Comment:
Does this need to be the same as the used Bazel version? If yes, is it possible to couple it in the code somehow?

Reply:
It should be the same, but I don't expect severe errors when being off by a minor version. Currently we are using `5.2.0`, but I decided to already use the latest `5.` version here so we don't have to touch this when we go to `5.3.x`.

Regarding coupling: It's probably not too hard, The version is defined in `$MAGMA_ROOT/.bazelversion`. If we require that `MAGMA_ROOT` is configured when this script is executed we can just read from there. I wouldn't try to reference the `.bazelversion` path relative to this script, because figuring out the path of the currently executed script is surprisingly difficult.

So far I was hoping I can postpone this question to when we actually do an upgrade ;) But I'm fine with doing it now. Thinking of it, the caller of the script already has to know the `MAGMA_ROOT`, so it's probably the cleanest approach if the caller figures out the correct Bazel version and then passes it to this script.